### ISSUE TITLE
slack-message.c:138:12: error: ‘PURPLE_MESSAGE_REMOTE_SEND’ undeclared

### ISSUE THREAD

**Issue by arne-cl** [dylex/slack-libpurple#3]
Hi Dylan,

when I try to build the software I get the following error. I used `libpurple-dev 1:2.10.12-0ubuntu5.2`.

```
~/bin/slack-libpurple $ make install
gcc -c -g -O2 -Wall -Werror -fPIC -D_DEFAULT_SOURCE=1 -std=c99 -I/usr/local/include/json-parser -I/usr/include/libpurple -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -o slack-message.o slack-message.c
slack-message.c: In function ‘handle_message’:
slack-message.c:138:12: error: ‘PURPLE_MESSAGE_REMOTE_SEND’ undeclared (first use in this function)
   flags |= PURPLE_MESSAGE_REMOTE_SEND;
            ^
slack-message.c:138:12: note: each undeclared identifier is reported only once for each function it appears in
Makefile:43: recipe for target 'slack-message.o' failed
make: *** [slack-message.o] Error 1
```

Best,
Arne

**Comment by dylex** [comment]
Apparently that flag was only added in libpurple 2.12, so if you have something older it won't work. I've added a check for this, now, though there might be other newer features you run into, as well.

**Comment by glajchs** [comment]
I'm having this same problem (though with a slightly different error message).  I solved it by updating to libpurple 2.12, but that's not always easy.  I followed this and it worked to upgrade me:
http://ubuntuhandbook.org/index.php/2017/03/how-to-install-pidgin-2-12-in-ubuntu-16-04-via-ppa/

**Comment by dylex** [comment]
If you want to post the error I might be able to fix it as well. I haven't tested on very many older versions.

**Comment by glajchs** [comment]
I had already upgraded and closed the terminal that had the error, but luckily I have the same OS version installed in a VM (for other testing) and was able to reproduce it there.  Output below in the gist

https://gist.github.com/glajchs/cb3b34bf565488493bdd8c508795bd64

**Comment by dylex** [comment]
Thanks, @glajchs.  If it's not too much trouble, could you try again with the latest code?

**Comment by glajchs** [comment]
Sure, the fix worked!  It built without any errors using purple 2.10 libraries!
Just to be complete, I tried to load pidgin + slack using said build, and it worked.

One annoyance was that it didn't work at first, and I figured out it was because the libjsonparser.so.1.1 wasn't getting picked up by default.  Running "ldd" on "/usr/lib/purple-2/libslack.so" found that "libjsonparser.so.1.1" wasn't found.  "locate" found that library at "/usr/local/lib", so I just defined an LD_LIBRARY_PATH on the command line (export LD_LIBRARY_PATH=/lib:/usr/lib:/usr/local/lib) then ran pidgin again, and the slack module showed up and worked.
I didn't require this extra step on my local machine (somewhere over the years I probably had included /usr/local/lib in the libraries path), but I did require this step on the VM I was running.  I use this VM as a pretty barebones machines for testing purposes.

I know it's not really your job to tell people how to install and use json-parser, but if there's an easier workaround to ensure the library shows up, you might want to mention it in the docs.

Thanks for the quick turnaround!  My company is moving from Hipchat to Slack in the coming days/weeks, so hopefully I can find some time to poke around and make a few improvements.
